### ISSUE TITLE
posterior_predict fails for binomial models with missing interactions

### ISSUE THREAD

**Issue by apkoenigsberg** [stan-dev/rstanarm#260]
#### Summary:
Running posterior_predict on the output of an rstanarm::stan_glm estimation when there are interaction terms with missing levels causes an error. This only happened when my model was binomial.

#### Description:
I have a model I'm estimating where I want an interaction term between two categorical variables. Not all combinations of both variables are present, so when rstanarm estimates it lets me know that it dropped those levels (e.g. `In center_x(x, sparse) : Dropped empty interaction levels: group1B:group2c`). When I go to run posterior_predict I get the error: 
```
Error in stanmat[, beta_sel, drop = FALSE] : subscript out of bounds
``` 
because model.matrix creates a column with the empty interaction level, but the parameter matrix doesn't.

#### Reproducible Steps:
```
library(rstanarm)

group1 <- c("A", "B")
group2 <- c("a", "b", "c")
testFrame <- expand.grid(group1 = group1, group2 = group2)[1:5,]
testFrame$n <- 100
testFrame$theta <- runif(5)
testFrame$e <- round(testFrame$n*testFrame$theta)

fit <- rstanarm::stan_glm(cbind(e, n - e) ~ group1:group2, data = testFrame, 
                          family = binomial("logit"))
pred <- rstanarm::posterior_predict(fit)
```

#### RStanARM Version:
> packageVersion("rstanarm")
[1] ‘2.17.2’

#### R Version:
> getRversion()
[1] ‘3.4.2’

#### Operating System:
SUSE Linux 11sp4

**Comment by groceryheist** [comment]
I also have this issue. Is there any work-around?

**Comment by bakaburg1** [comment]
same here

**Comment by jgabry** [comment]
Sorry this got left behind for so long! Should be fixed by PR #522.
